pkg/apis/serving/v1alpha1/configuration_lifecycle.go

```diff
@@ -79,18 +79,27 @@ func (cs *ConfigurationStatus) InitializeConditions() {
 func (cs *ConfigurationStatus) SetLatestCreatedRevisionName(name string) {
 	cs.LatestCreatedRevisionName = name
 	if cs.LatestReadyRevisionName != name {
-		confCondSet.Manage(cs).MarkUnknown(
-			ConfigurationConditionReady,
-			"",
-			"")
+		cs.MarkConfigurationUnknown()
 	}
 }
 
 func (cs *ConfigurationStatus) SetLatestReadyRevisionName(name string) {
 	cs.LatestReadyRevisionName = name
+}
+
+// MarkConfigurationReady sets the Ready state of a configuration to True
+func (cs *ConfigurationStatus) MarkConfigurationReady() {
 	confCondSet.Manage(cs).MarkTrue(ConfigurationConditionReady)
```

Comment:
This seems like a huge change.  Why can't we just guard this in a check that `LatestCreated == LatestReady`?